server/util/terminal_writer/terminal_writer.go

```diff
@@ -0,0 +1,26 @@
+package terminal_writer
+
+import (
+	"github.com/buildkite/terminal-to-html/v3"
+)
+
+type Writer struct {
+	screen *terminal.Screen
+}
+
+func NewWriter() *Writer {
+	return &Writer{screen: terminal.NewScreen()}
+}
+
+func (w *Writer) Write(p []byte) (int, error) {
+	terminal.ParseANSIToScreen(w.screen, p)
+	return len(p), nil
+}
+
+func (w *Writer) Render() []byte {
+	return w.screen.AsANSI()
+}
+
+func (w *Writer) PopExtraLines(numLinesToRetain int) []byte {
+	return w.screen.FlushLinesFromTop(numLinesToRetain)
+}
```

Comment:
nit: if we rename this back to `terminal.go` and then also rename it back to ScreenWriter, the diff in this PR would be a lot smaller

Reply:
I also like the "terminal" package name better, because it provides both read/write functionality

Reply:
Sure, can do.